subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2920,11 +2936,15 @@ static void le_df_connectionless_iq_report(struct pdu_data *pdu_rx,
 	}
 #endif /* CONFIG_BT_CTLR_DF_SCAN_CTE_RX */
 
-	/* If there are no IQ samples due to insufficient resources
-	 * HCI event should inform about it by storing single octet with
-	 * special I_sample and Q_sample data.
+	/* If packet status does not indicate insufficient resources for IQ samples and for
+	 * some reason sample_count is zero, inform Host about lack of valid IQ samples by
+	 * storing single I_sample and Q_sample with BT_HCI_LE_CTE_REPORT_NO_VALID_SAMPLE value.
 	 */
-	samples_cnt = (!iq_report->sample_count ? 1 : iq_report->sample_count);
+	if (iq_report->packet_status == BT_HCI_LE_CTE_INSUFFICIENT_RESOURCES) {
+		samples_cnt = 0
```

Comment:
```suggestion
		samples_cnt = 0;
```

Reply:
Ups... I should have spot that! Thank you!

Reply:
Did it pass CI before being fixed? Are these being built?

Reply:
There were other errors in CI builds that could cover this one.
Also the DF tests/code are not always executed by CI. I've got a WPI PR to fix that: https://github.com/zephyrproject-rtos/zephyr/pull/45333